src/consensus/aft/raft.h

```diff
@@ -533,14 +537,45 @@ namespace aft
                       .get_current_time_offset();
         request_tracker->tick(time);
 
-        if (is_follower() && has_bft_timeout_occurred(time))
+        if (
+          is_follower() &&
+          (state->view_change_in_progress || has_bft_timeout_occurred(time)) &&
+          view_change_tracker.should_send_view_change(time))
         {
+          state->view_change_in_progress = true;
```

Comment:
Seems odd that this bit of state isn't delegated to the tracker, which seems to also know that there is a pending view change?

Reply:
thank you, I missed deleting this :) the tracker does all this as it should